### PR TITLE
CopterHack-2022: Edgenoon Team

### PR DESCRIPTION
**Pull request by bartoszptak**

### CODE REVIEW

File: docs/en/neural_vision_based_landing_method.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Neural and vision-based landing method
+
+[CopterHack-2022](copterhack2022.md), team Edgenoon.
+
+## Team information
+
+The list of team members:
+
+* Bartosz Ptak, bartptak@gmail.com / @latajacy6, engineer and programmer.
+* Mateusz Piechocki, piechocki.mateusz97@gmail.com , engineer and programmer.
```
Comment (okalachev) [inline docs/en/neural_vision_based_landing_method.md:10]:
```suggestion
* Mateusz Piechocki, piechocki.mateusz97@gmail.com, engineer and programmer.
```

File: docs/en/neural_vision_based_landing_method.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+## Project description
+
+![raspberry-and-ncs2](../assets/rpi_ncs2.jpg)
+
+### Project idea
+
+The project involves increasing the precision of landing (one of the most
+dangerous maneuvers for flying machines) under operational conditions on a
+ mobile platform and taking care of safety in its vicinity.
```
Comment (okalachev) [inline docs/en/neural_vision_based_landing_method.md:20]:
```suggestion
The project involves increasing the precision of landing (one of the most dangerous maneuvers for flying machines) under operational conditions on a mobile platform and taking care of safety in its vicinity.
```

File: docs/en/neural_vision_based_landing_method.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+Due to these reasons, we plan to implement autonomous landing on a specifically
+designed pad that is marked with graphical elements, making it possible to recover
+its relative pose and orientation. Additional safety measures will be
+implemented - no landing is attempted if persons are present in the vicinity of the
+landing pad. We want to achieve this using convolutional neural networks and USB
+inference accelerators, for example, Neural Compute Stick 2 or Google Coral USB Accelerator.
```
Comment (okalachev) [inline docs/en/neural_vision_based_landing_method.md:27]:
```suggestion
Due to these reasons, we plan to implement autonomous landing on a specifically designed pad that is marked with graphical elements, making it possible to recover its relative pose and orientation. Additional safety measures will be implemented - no landing is attempted if persons are present in the vicinity of the landing pad. We want to achieve this using convolutional neural networks and USB inference accelerators, for example, Neural Compute Stick 2 or Google Coral USB Accelerator.
```

File: docs/en/neural_vision_based_landing_method.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+### The potential outcomes
+
+Precision landing is essential for building mobile flying robots. The use of classical
+GPS or very costly GPS RTK allows you to move closer to the landing platform with
+limited precision, while a vision-based system using deep neural networks will enable
+not only precise landing without the need for tags, but will also take care of safety
+in its area.
```
Comment (okalachev) [inline docs/en/neural_vision_based_landing_method.md:35]:
```suggestion
Precision landing is essential for building mobile flying robots. The use of classical GPS or very costly GPS RTK allows you to move closer to the landing platform with limited precision, while a vision-based system using deep neural networks will enable not only precise landing without the need for tags, but will also take care of safety
in its area.
```

File: docs/en/neural_vision_based_landing_method.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+### Using Clover platform
+
+The clover platform will be used as the basis for the safe testing of landing algorithms
+in a real-world environment.
```
Comment (okalachev) [inline docs/en/neural_vision_based_landing_method.md:40]:
```suggestion
The clover platform will be used as the basis for the safe testing of landing algorithms in a real-world environment.
```

File: docs/en/neural_vision_based_landing_method.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+### Additional information at the request of participants
+
+We are fresh Ph.D. students at Poznan University of Technology. We have successfully
+developed an intelligent system based on a drone with an object detection algorithm
+for waste detection. The system in 2021 was awarded 3rd prize during the Teknofest
+festival’s “Free mission” contest in Turkey. Project website: [UAVVaste](https://uavvaste.github.io/).
```
Comment (okalachev) [inline docs/en/neural_vision_based_landing_method.md:47]:
```suggestion
We are fresh Ph.D. students at Poznan University of Technology. We have successfully developed an intelligent system based on a drone with an object detection algorithm for waste detection. The system in 2021 was awarded 3rd prize during the Teknofest festival’s “Free mission” contest in Turkey. Project website: [UAVVaste] https://uavvaste.github.io/).
```

**Comment by okalachev** [comment]
Hello! Note, that Clover documentation doesn't have a line width limit in markdown file, we use "infinite" lines and line wrapping instead.

**Comment by bartoszptak** [comment]
@okalachev Thank you for the corrections. Regarding the line length, we were using the default rules and that's why the discrepancy.

File: docs/en/neural_vision_based_landing_method.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
+# Neural and vision-based landing method
+
+[CopterHack-2022](copterhack2022.md), team Edgenoon.
+
+## Team information
+
+The list of team members:
+
+* Bartosz Ptak, bartptak@gmail.com / @latajacy6, engineer and programmer.
```
Comment (okalachev) [inline docs/en/neural_vision_based_landing_method.md:9]:
```suggestion
* Bartosz Ptak, bartptak@gmail.com / [@latajacy6](https://t.me/latajacy6), engineer and programmer.
```

**Comment by SeliverstovaE** [comment]
Dear friend,

Thank you for your interest in the CopterHack 2022 contest. Your application has successfully passed the experts’ review and was accepted for the second project stage of the competition, which will be held until February 28, 2022. This stage will be followed by monthly updates, lectures, and guidance from industry experts.

What's next?

1. Join the Telegram group with participants of the second project stage “[CopterHack 2022, participants chat](https://t.me/joinchat/1KIDu8p5FGhlYTBi )”.  
2. Every month, you're expected to prepare free-form reports on the progress of your project (update) in this draft pull request. In your report please indicate your progress, what tasks have been solved. You are encouraged to attach photos and videos that demonstrate the results of the work done within the month.
3. Send the link to the update to the participants chat in the Telegram group.

In response to the message, you will receive feedback from the competition jury.

There is no specific date for sending the report, so you can send the report between the beginning and the end of the month.

Monthly updates become mandatory from November 2021. But as soon as you are in the telegram group, you can already send updates and receive feedback from the jury.

We are interested in helping you to complete your project as much as possible by the final stage. We wish you success!

Always in touch! Your CopterHack team.

**Comment by bartoszptak** [comment]
# November update

Hi, we prepared an update on our project. 

## Tags
Event: **Copterhack2022**
Team: **Edgenoon**
Data: **29 November 2021**

## What's done
- [x] We researched the performance of small neural networks for object detection using Raspberry Pi 4 and Intel Neural Computer Stick 2. We prepared [a table](https://github.com/edgenoon-ai/VITAL-clover#preliminary-algorithm-benchmarks) with results.
- [x] We built a docker image for the Clover simulation based on ROS-Noetic and PX4. [It's here](https://github.com/edgenoon-ai/VITAL-clover/tree/main/docker).
- [x]  We tested in the simulation the commands to move and to update the home position.

## Next steps

- [ ] We are waiting for the delivery of the Clover drone. We will need to assemble it and familiarize ourselves with the functionality.
- [ ] We need to design a vector-based and non-symmetric helipad and print it. 
- [ ] We need to explore the camera properties at the raspberry and collect data.

**Comment by bartoszptak** [comment]
# December update

Hi, we prepared an update on our project. 

## Tags
Event: **Copterhack2022**
Team: **Edgenoon**
Data: **31 December 2021**

## What's done
- [x] After some additional testing and the change of inference framework to [onnxruntime with openvino provider](https://onnxruntime.ai/docs/execution-providers/OpenVINO-ExecutionProvider.html), we decided to use the NanoDet-t model.
- [x] We designed [the helipad](https://github.com/edgenoon-ai/VITAL-clover/blob/main/helipad/helipad.svg). It is in vector format and allows any scaling.
- [x] We have received the Clover, assembled it and [performed a few flying tests](https://www.youtube.com/watch?v=PU_oj93jJds).
- [x] We compared the Clover camera and the RPi 2 and [performed some inference](https://www.youtube.com/watch?v=dZroPemLW18) on a model trained on the Visdrone dataset. 

## Next steps

- [ ] Designing a scheme of the whole system.
- [ ] Implementation of the system in simulation.
- [ ] Collection of data for the algorithm.
- [ ] More tests of Clover.

**Comment by SeliverstovaE** [comment]
Dear Edgenoon Team,
We are closing pull request for your project as you didn’t get to the final round of the competition. 

You are welcome to submit your article to the clover documentation as an clover platform-based project. In order to do this please create a new pull request or make a comment to the existing one.

We wish you success in implementing your project and invite you to the next events.

**Pull request closed without merging**